### DISCUSSION TITLE
Slicing a 3D object into small cube cells

### DISCUSSION THREAD

**Discussion by AsimAl-Sofi** [pyvista/pyvista#4238] [Q&A]
Hello
Iam currently working on a project where I need to divide a 3d object into small cube cells (of same size when possible).
For each cube cell I need the coordinates of the center of the cube cell as well as the center points of each surface of the cube cell together with its surface value.

Any suggestions/ideas how to achieve this?
Thanks 
<img width="366" alt="DivideInCells" src="https://user-images.githubusercontent.com/130053977/230355715-bf21986b-8154-4776-a5ca-77bb1d8e6e89.PNG">

**Comment by banesullivan** [comment]
Are you trying to partition the mesh for HPC, parallel processing, or analysis? Or do you simply want to chop up the mesh into these cubes?

**Reply by AsimAl-Sofi** [reply]
Hi
Iam not familiar with HPC parallel processing, so actually what I want is to chop the mesh into small cube cells and after that do some calculations on each cell.

**Reply by akaszynski** [reply]
HPC would be for distributing a large mesh over several clusters. There is a [partition](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSetFilters.partition.html) filter that you might use that works fairly well. This is well suited for the HPC case. There are other solutions for data partitioning.

**Reply by AsimAl-Sofi** [reply]
Thank you @akaszynski for the explanation
this helps alot
However, Iam still struggling extracting the coordinates of the center points of the cell and of the faces.
For example, how to get the coordinates of the first cell of the first cluster, i.e., out[0].get_cell(0). And how to get the surface of each face of the cell?
